windows/hid.c

```diff
@@ -164,30 +140,47 @@ static struct hid_api_version api_version = {
 	typedef CONFIGRET(__stdcall* CM_Get_Parent_)(PDEVINST pdnDevInst, DEVINST dnDevInst, ULONG ulFlags);
 	typedef CONFIGRET(__stdcall* CM_Get_DevNode_PropertyW_)(DEVINST dnDevInst, CONST DEVPROPKEY* PropertyKey, DEVPROPTYPE* PropertyType, PBYTE PropertyBuffer, PULONG PropertyBufferSize, ULONG ulFlags);
 	typedef CONFIGRET(__stdcall* CM_Get_Device_Interface_PropertyW_)(LPCWSTR pszDeviceInterface, CONST DEVPROPKEY* PropertyKey, DEVPROPTYPE* PropertyType, PBYTE PropertyBuffer, PULONG PropertyBufferSize, ULONG ulFlags);
+	typedef CONFIGRET(__stdcall* CM_Get_Device_Interface_List_SizeW_)(PULONG pulLen, LPGUID InterfaceClassGuid, DEVINSTID_W pDeviceID, ULONG ulFlags);
+	typedef CONFIGRET(__stdcall* CM_Get_Device_Interface_ListW_)(LPGUID InterfaceClassGuid, DEVINSTID_W pDeviceID, PZZWSTR Buffer, ULONG BufferLen, ULONG ulFlags);
 
 	static CM_Locate_DevNodeW_ CM_Locate_DevNodeW = NULL;
 	static CM_Get_Parent_ CM_Get_Parent = NULL;
 	static CM_Get_DevNode_PropertyW_ CM_Get_DevNode_PropertyW = NULL;
 	static CM_Get_Device_Interface_PropertyW_ CM_Get_Device_Interface_PropertyW = NULL;
+	static CM_Get_Device_Interface_List_SizeW_ CM_Get_Device_Interface_List_SizeW = NULL;
+	static CM_Get_Device_Interface_ListW_ CM_Get_Device_Interface_ListW = NULL;
 
 	static HMODULE cfgmgr32_lib_handle = NULL;
 #endif /* HIDAPI_USE_DDK */
 
+#undef MIN
+#define MIN(x,y) ((x) < (y)? (x): (y))
+
+/* MAXIMUM_USB_STRING_LENGTH from usbspec.h is 255 */
+/* BLUETOOTH_DEVICE_NAME_SIZE from bluetoothapis.h is 256 */
+#define MAX_STRING_WCHARS 256
+
+static struct hid_api_version api_version = {
+	.major = HID_API_VERSION_MAJOR,
+	.minor = HID_API_VERSION_MINOR,
+	.patch = HID_API_VERSION_PATCH
+};
+
 struct hid_device_ {
-		HANDLE device_handle;
-		BOOL blocking;
-		USHORT output_report_length;
-		unsigned char *write_buf;
-		size_t input_report_length;
-		USHORT feature_report_length;
-		unsigned char *feature_buf;
-		void *last_error_str;
-		DWORD last_error_num;
-		BOOL read_pending;
-		char *read_buf;
-		OVERLAPPED ol;
-		OVERLAPPED write_ol;
-		struct hid_device_info* device_info;
+	HANDLE device_handle;
+	BOOL blocking;
+	USHORT output_report_length;
+	unsigned char *write_buf;
+	size_t input_report_length;
+	USHORT feature_report_length;
+	unsigned char *feature_buf;
+	void *last_error_str;
+	DWORD last_error_num;
+	BOOL read_pending;
+	char *read_buf;
+	OVERLAPPED ol;
+	OVERLAPPED write_ol;
+	struct hid_device_info* device_info;
```

Comment:
None of it is relevant to the topic of the PR.
Please revert all the white-space changes and shuffling around of the order of declarations.

It makes _unnecessary noise in the git history_, which I often need to consult.

Reply:
alright